src/m365/spo/commands/file/file-sharinglink-list.ts

```diff
@@ -0,0 +1,156 @@
+import { AxiosRequestConfig } from 'axios';
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { formatting } from '../../../../utils/formatting';
+import { validation } from '../../../../utils/validation';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  fileId?: string;
+  fileUrl?: string;
+}
+
+export interface GraphFileDetails {
+  SiteId: string;
+  VroomDriveID: string;
+  VroomItemID: string;
+}
+
+class SpoFileSharinglinkListCommand extends SpoCommand {
+  public get name(): string {
+    return commands.FILE_SHARINGLINK_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists all the sharing links of a specific file';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'roles', 'link'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        fileId: typeof args.options.fileId !== 'undefined',
+        fileUrl: typeof args.options.fileUrl !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --webUrl <webUrl>'
+      },
+      {
+        option: '-i, --fileId [fileId]'
+      },
+      {
+        option: '-f, --fileUrl [fileUrl]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.webUrl);
+        if (isValidSharePointUrl !== true) {
+          return isValidSharePointUrl;
+        }
+
+        if (args.options.fileId) {
+          if (!validation.isValidGuid(args.options.fileId)) {
+            return `${args.options.fileId} is not a valid GUID`;
+          }
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['fileId', 'fileUrl'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Retrieving sharing links for file ${args.options.fileId || args.options.fileUrl}...`);
+    }
+
+    try {
+      const fileDetails = await this.getFileDetails(args.options.webUrl, args.options.fileId, args.options.fileUrl);
+      const sharingLinks = await this.getFileSharingLinks(fileDetails);
+
+      if (!args.options.output || args.options.output === 'json') {
```

Comment:
The issue currently is that it will show ## undefined (ID of the sharing link) as the response object does not contain a title. This is why I excluded this from the doc file.

Reply:
What I meant is, right now you are changing the JSON object so it is properly displayed for `text` and `csv` output. But in this case, `md` output will also get the stripped version as output, which is not what we want according to me.